chrome/content/zotero/bibliography.js

```diff
@@ -195,20 +203,55 @@ var Zotero_File_Interface_Bibliography = new function() {
 		
 		// Change label to "Citation" or "Note" depending on style class
 		if(document.getElementById("citations")) {
+			let label = "";
 			if(Zotero.Styles.get(selectedStyle).class == "note") {
-				var label = Zotero.getString('citation.notes');
+				label = Zotero.getString('citation.notes');
 			} else {
-				var label = Zotero.getString('citation.citations');
+				label = Zotero.getString('citation.citations');
 			}
 			document.getElementById("citations").label = label;
 		}
 
 		window.sizeToContent();
+	};
+
+	/*
+	 * Update locale menulist when style is changed
+	 */
+	function updateLocaleMenu(selectedStyle) {
+		// For styles with a default-locale, disable locale menulist and show locale
+		var menulist = document.getElementById("locale-menu");
+		
+		// If not null, then menulist is extended with the default-locale value
+		// of the previously selected style
+		if (defaultStyleLocale) {
+			// Reset menulist
+			menulist.removeItemAt(0);
+			defaultStyleLocale = "";
+		}
+		
+		if (selectedStyle.locale) {
+			defaultStyleLocale = selectedStyle.locale;
```

Comment:
I guess we could look up whether the parent has a default-locale, and set the dropdown to that locale if it exists. Not sure that's worth the trouble, since most our styles (dependent and independent) have a default-locale.